coverage/PENDING_REVIEW/intel/0454-COVERAGE-nir-lower-vars-to-ssa-624/cov-write-past-matrix-elements-unused.amber

```diff
@@ -0,0 +1,236 @@
+#!amber
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# A test for a coverage-gap found by the GraphicsFuzz project.
+
+# Short description: A fragment shader that covers specific NIR code paths
+
+# The test passes because the shader always writes red.
+
+# Optimized using spirv-opt with the following arguments:
+# '--reduce-load-size'
+# '--if-conversion'
+# '--if-conversion'
+# '--combine-access-chains'
+# '--eliminate-dead-branches'
+# '--merge-return'
+# '--eliminate-dead-branches'
+# '--merge-blocks'
+# '--if-conversion'
+# '--eliminate-local-single-store'
+# '--redundancy-elimination'
+# '--private-to-local'
+# '--combine-access-chains'
+# '--ccp'
+# '--eliminate-local-multi-store'
+# '--if-conversion'
+# '--convert-local-access-chains'
+# '--combine-access-chains'
+# '--reduce-load-size'
+# '--ccp'
+# '--eliminate-dead-inserts'
+# '--simplify-instructions'
+# '--eliminate-dead-code-aggressive'
+# spirv-opt commit hash: a0370efd589be33d5d9a85cfde2f85841b3755af
+
+
+
+SHADER vertex variant_vertex_shader PASSTHROUGH
+
+# variant_fragment_shader is derived from the following GLSL:
+# #version 320 es
+# #define _int_0 _GLF_uniform_int_values[0]
+# #define _int_1 _GLF_uniform_int_values[1]
+# #define _int_2 _GLF_uniform_int_values[2]
+# #define _float_1_0 _GLF_uniform_float_values[0]
+# #define _float_0_0 _GLF_uniform_float_values[1]
+#
+# precision highp int;
+# precision highp float;
+#
+# // Contents of _GLF_uniform_int_values: [0, 1, 2]
+# layout(set = 0, binding = 0) uniform buf0
+# {
+#     int _GLF_uniform_int_values[3];
+# };
+# // Contents of _GLF_uniform_float_values: [1.0, 0.0]
+# layout(set = 0, binding = 1) uniform buf1
+# {
+#     float _GLF_uniform_float_values[2];
+# };
+#
+# layout(location = 0) out vec4 _GLF_color;
+#
+# void main()
+# {
+#     mat3x2 m32 = mat3x2(_float_1_0);
+#     int a = 3;
+#     // Indexing past the matrix elements, but the matrix
+#     // values are not used for the final results.
+#     m32[a][_int_0] = _float_1_0;
+#     float sums[3] = float[3](_float_1_0, _float_1_0, _float_1_0);
```

Comment:
As per @paulthomson's comment on a different PR, we want to avoid an OOB access even if its result is not used. Can you either clamp it or make it unreachable?

Reply:
Updated.